sys/include/net/gnrc/ppp/fsm.h

```diff
@@ -0,0 +1,291 @@
+    void (*on_layer_down)(gnrc_ppp_fsm_t *cp);                           /**< Optional callback when current FSM is down */
+} gnrc_ppp_fsm_t;
+
+/**
+ * @brief Data type or representing a PPP option
+ */
+typedef struct gnrc_ppp_fsm_conf {
+    uint8_t type;                   /**< Option type */
+    network_uint32_t value;         /**< Current option value */
+    network_uint32_t default_value; /**< Default value of option (for resetting purposes). */
+    size_t size;                    /**< Number of bytes of current option */
+                                    /**
+                                     * @brief flags of the current options.
+                                     *
+                                     * Set OPT_ENABLED flag for enabling or disabling the option. Set OPT_REQUIRED flag if option is mandatory
+                                     */
+    uint8_t flags;
+                                    /**
+                                     * @brief hook for is_valid function for a received option in a Configure Request
+                                     *
+                                     * @details This hook should return true if the option is valid. False otherwise
+                                     */
+    uint8_t (*is_valid)(gnrc_ppp_option_t *opt);
+                                    /**
+                                     * @brief hook for build_nak_options function for a NAK'd option in a Configure Request
+                                     *
+                                     * @details This hook should write the suggested options in buf. If buf is NULL, returns the size of the options.
+                                     * Always returns the size of written buf.
+                                     */
+    uint8_t (*build_nak_opts)(uint8_t *buf);
+                                    /**
+                                     * @brief hook for set function for a ACK'd option in a Configure Request
+                                     *
+                                     * @details This hook should do whatever is necessary after ACK'd an option.
+                                     */
+    void (*set)(gnrc_ppp_fsm_t *t, gnrc_ppp_option_t *opt, uint8_t peer);
+    gnrc_ppp_fsm_conf_t *next;        /**< pointer to next configuration */
+} gnrc_ppp_fsm_conf_t;
+
+/**
+ * @brief init the Option Negotiation Automaton FSM
+ *
+ * @param[in] ppp_dev pointer to GNRC pppdev object
+ * param[in] cp Pointer to the Option Negotiation Automaton FSM
+ *
+ * @return 0 for now.
+ */
+int fsm_init(gnrc_netdev2_t *ppp_dev, gnrc_ppp_fsm_t *cp);
+/**
+ * @brief triggers an event in the FSM
+ *
+ * @param cp Pointer to FSM
+ * @param event event of Option Negotiation Automaton
+ * @param pkt Pointer to a pkt. NULL if the event is not packet related
+ *
+ * @return -EBADMSG if there was an error in the packet
+ * @return 0
+ */
+int trigger_fsm_event(gnrc_ppp_fsm_t *cp, int event, gnrc_pktsnip_t *pkt);
```

Comment:
And also, for naming consistency with the header I would recommend `gnrc_ppp_fsm_trigger_event()`